sparkplug/config/connection.py

```diff
@@ -59,28 +59,33 @@ def __init__(self, name, channel_configurer, reconnect_delay='10', **kwargs):
 
         self.channel_configurer = channel_configurer
 
-    def run_channel(self, channel):
+    def run_channel(self, connection, channel):
         _log.debug("Configuring channel elements.")
         self.channel_configurer.start(channel)
         try:
-            self.pump(channel)
+            self.pump(connection, channel)
         except (SystemExit, KeyboardInterrupt):
             _log.debug("Tearing down connection.")
             self.channel_configurer.stop(channel)
             raise
 
-    def pump(self, channel):
+    def pump(self, connection, channel):
         while True:
             _log.debug("Waiting for a message.")
-            channel.wait(spec.Basic.Deliver)
+            try:
+                channel.wait(spec.Basic.Deliver, timeout=connection.heartbeat)
+            except socket.timeout :
```

Comment:
From the looks, this will swallow all socket timeouts, right? It it worth logging the exception?

Reply:
Sorry, I missed this comment before I merged.   I'm not sure what our temperature is for logging volume (i.e. will show a timeout every connection.heartbeat seconds), but I can go back and add that logger in if you think it's important.

Reply:
Actually, there's no need to log it since the exception is as a result of the timeout waiting on the channel.